packages/text-select/src/token-select/index.jsx

```diff
@@ -43,54 +44,82 @@ export class TokenSelect extends React.Component {
     );
   };
 
-  toggleToken = (index, t) => {
-    const { onChange, maxNoOfSelections } = this.props;
-    const selected = !t.selected;
-    if (
-      selected &&
-      maxNoOfSelections > 0 &&
-      this.selectedCount() >= maxNoOfSelections
-    ) {
-      log('skip toggle max reached');
-      return;
-    }
-    const update = { ...t, selected: !t.selected };
+  toggleToken = ({ target }) => {
+    const { showCorrectnessToken } = this.props;
     const tokens = clone(this.props.tokens);
-    tokens.splice(index, 1, update);
-    onChange(tokens);
+    /* We take the first parent element that has spanWrapper class
+      spanWrapper class is a class that I added on each span that wraps a token
+      (Token & CorrectnessToken)
+    */
+    const span = target.closest(`.${Token.rootClassName}`);
+
+    /* indexkey is an attribute that I added on each span that wraps a token
```

Comment:
move all comments to above the function and use js doc style: 
```javascript
/**
 * Here is the comment
 */
```
Keep the comment short, only describe what is non-standard (aka having to extract `indexkey`). Don't use 'I' or 'we'. 
In general, try and avoid having to add comments in favour of making the code explain what is happening within. because what you're doing is non standard i think a little explanation is ok.

Reply:
Thank you for the review, @edeustace . I updated my code, please let me know if I still have to fix something.

Reply:
just need to update the test snapshots - cheers